Formula/pnpm.rb

```diff
@@ -20,14 +20,66 @@ class Pnpm < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "52d1cbcba6a433d4f18eef3d3efafe44ac46b41fcbbc6be1956109ce69cb305f"
   end
 
-  depends_on "node"
+  depends_on "node@14" => :build
+
+  # Described in https://github.com/pnpm/pnpm#installation
+  # Managed by https://github.com/pnpm/get
+  resource "pnpm-buildtime" do
+    url "https://get.pnpm.io/v6.7.js"
+    sha256 "56bffe269aab055bec3a07fa65c387f5a35729550dc7fc76711844a2de6e1b8b"
+  end
 
   def install
-    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
-    bin.install_symlink Dir["#{libexec}/bin/*"]
+    buildtime_bin = buildpath/"buildtime-bin"
+    resource("pnpm-buildtime").stage do |r|
+      buildtime_bin.install "v#{r.version}.js"
+      (buildtime_bin/"pnpm").write_env_script "#{Formula["node@14"].bin}/node",
+        buildpath/"buildtime-bin/v#{r.version}.js", {}
+      chmod 0755, buildtime_bin/"pnpm"
+    end
+    ENV.prepend_path "PATH", buildtime_bin
+    system "pnpm", "install"
+    chdir "packages/pnpm" do
+      system buildpath/"node_modules/.bin/tsc", "--build"
+      system "pnpm", "run", "bundle"
+      on_macos do
+        system "node_modules/.bin/pkg", "--target=host", "--out-path=dist", "dist/pnpm.cjs"
+      end
+      on_linux do
+        system "node_modules/.bin/pkg", "--target=linuxstatic", "--out-path=dist", "dist/pnpm.cjs"
```

Comment:
If this makes use of static libraries rather than dynamic I don't think we want this.

Reply:
There is a problem using dynamic libraries on Linuxbrew. I think it cannot be circumvented easily.
https://github.com/Homebrew/homebrew-core/pull/82892#issuecomment-894904612

Reply:
Then that should be fixed, not worked around.

CC @Homebrew/linux

Reply:
I couldn't fix this problem on Linuxbrew.
I expected that no libraries are relocated with any_skip_relocation, but the actual was that the path of `/lib64/ld-linux-x86-64.so.2` was relocated with the bottle installation.